### ISSUE TITLE
"Rename" buttons

### ISSUE THREAD

**Issue by NecroBones** [KSP-SpaceDock/SpaceDock#62]
Something that KerbalStuff was always missing was an option to rename things. I know Sircmpwn had his reasons for not allowing renames of mods. Do we want to revisit this?

Also, the "mod packs" lack a rename option (as well as the ability to be deleted by the owner). Perhaps the packs should be able to be renamed, whether the mods can be renamed or not?

**Comment by phmayo** [comment]
The packs can be deleted, see #55 (no clue if that is in prod yet, though).

We would have to investigate the impact of renaming things. if the names become canonical for CKAN (which is what it looks like, depending on what "name" we are talking about), the effort might be larger than the gain.

**Comment by NecroBones** [comment]
Not in prod, last I looked this morning. But yeah, I can see where that would be an issue with canonical names. But then again, the NetKan configs link via the mod's ID number, so it may not be as big of an issue as we think, depending on how that is implemented.

**Comment by phmayo** [comment]
Nah, the CKAN specs refer to a canonical name used by CKAN to determine "recommends" and "depends", so name changes may have seriously unintended side-effects. and it's not something actually unique, like a GUID.

And since we are stuck with testing in prod ATM, a fix will take a while, alas.

**Comment by NecroBones** [comment]
No, I mean the actual "kref" line in the netkan file points to a GUID. The canonical name appears to then be internal to CKAN, and differs from the naming convention for the (what used to be) KerbalStuff's naming convention. For instance KS would insert underscores in place of spaces, whereas on the CKAN side it's preferable to use concatenated names with the whitespace stripped.

**Comment by phmayo** [comment]
Oh, I see. Yeah, that would be easier to pull off.

**Comment by NecroBones** [comment]
It's still a valid concern though, and we'd have to get the CKAN folks involved to be sure it wouldn't hose anything up. But on the surface it looks doable.

**Comment by plague006** [comment]
It wouldn't break anything on CKAN's side. For us `name` is strictly an aesthetic feature for users. We use `identifier` as unique IDs for mods and as @NecroBones pointed out we use SpaceDock's mod ID as a target when scraping for info. The only time `name` is an issue for CKAN is when 2 mods share an identical name, as only 1 mod with a given name will be listed in the GUI.

**Comment by HebaruSan** [comment]
> I know Sircmpwn had his reasons for not allowing renames of mods.

Just in case, I checked [KerbalStuff's issues list](https://github.com/KerbalStuff/KerbalStuff/issues?q=is%3Aissue+) and found one example: 

https://github.com/KerbalStuff/KerbalStuff/issues/48#issuecomment-55700169

> This is intentionally not permitted, but if you send an email to support@kerbalstuff.com and explain your circumstances, it may be changed for you.

@ddevault didn't quite explain the reason, but it sounds like it was less of an architectural limitation than a rule, like changing mod names would somehow be disruptive or prone to abuse. I'm not really seeing the logic behind that, though, since it's easy enough to start a new mod with whatever name you like, and it definitely is annoying to be stuck with a flawed name. I've even see users delete mods and re-create them just to fix one or two letters in a name.
